src/app/qgsbookmarks.cpp

```diff
@@ -103,7 +107,9 @@ QgsBookmarks::QgsBookmarks( QWidget *parent ) : QDockWidget( parent )
 
 QgsBookmarks::~QgsBookmarks()
 {
-  delete lstBookmarks->model();
+  delete mQgisModel;
+  delete mProjectModel;
+  delete mModel;
```

Comment:
I would prefer QScopedPointer here: it clearly documents that the class is the owner and that it will be deleted during destruction